src/components/Contacts/ContactDetails/ContactDetailsTab/ContactDetailsTab.tsx

```diff
@@ -65,89 +118,189 @@ export const ContactDetailsTab: React.FC<ContactDetailTabProps> = ({
   });
 
   const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+  const { push, query } = useRouter();
+  const [deleteContact, { loading: deleting }] = useDeleteContactMutation();
+  const [dialogOpen, changeDialogOpen] = useState(false);
+
+  const { searchTerm } = query;
+
+  const handleDeleteContact = () => {
+    try {
+      deleteContact({
+        variables: {
+          accountListId,
+          contactId,
+        },
+        update: (cache, { data: deletedContactData }) => {
+          const deletedContactId = deletedContactData?.deleteContact?.id;
+          const query = {
+            query: ContactsDocument,
+            variables: {
+              accountListId,
+              searchTerm,
+            },
+          };
+
+          const dataFromCache = cache.readQuery<ContactsQuery>(query);
+
+          if (dataFromCache) {
+            const data = {
+              contacts: {
+                ...dataFromCache.contacts,
+                nodes: [
+                  ...dataFromCache.contacts.nodes.filter(
+                    (contact) => contact.id !== deletedContactId,
+                  ),
+                ],
+              },
+            };
+            cache.writeQuery({ ...query, data });
+
+            push({
+              pathname: '/accountLists/[accountListId]/contacts',
+              query: { accountListId, searchTerm },
+            });
+            enqueueSnackbar(t('Contact successfully deleted'), {
+              variant: 'success',
+            });
+          }
+        },
+      });
+    } catch (error) {
+      enqueueSnackbar(error.message, { variant: 'error' });
+      throw error;
+    }
+  };
+
+  const renderDeleteContactDialog = () => {
+    return (
+      <Dialog
+        open={dialogOpen}
+        aria-labelledby={t('Delete Contact Dialog')}
+        fullWidth
+        maxWidth="sm"
+      >
+        <DeleteContactTitle>
+          {t('Delete Contact')}
+          <CloseButton onClick={() => changeDialogOpen(false)}>
+            <CloseIcon titleAccess={t('Close')} />
+          </CloseButton>
+        </DeleteContactTitle>
+        <DialogContent dividers>
+          <DialogContentText>
+            {t(
+              'Are you sure you want to permanently delete this contact? Doing so will permanently delete this contacts information, as well as task history. This cannot be undone. If you wish to keep this information, you can try hiding this contact instead.',
```

Comment:
Do you guys keep long strings like this in a constant file or something? I started doing this so they could all be in one spot just in case wording needed to get changed.

Reply:
I believe this would live in a translations.json file and would have a specific key to match it with. But I'm not sure if we have that set up yet.

Reply:
Tatai started this with some sort of extraction tooling in mind.
https://github.com/CruGlobal/mpdx-react/blob/main/package.json#L19
https://github.com/CruGlobal/mpdx-react/blob/main/i18next-parser.config.js

Idk if it's one of these or something else https://www.i18next.com/overview/plugins-and-utils#extraction-tools. I haven't tried running any of it yet.

I think it's similar to gettext where it just goes looking for every string inside `t('<string>')` and extracts those into a file to upload. That way all the strings are colocated with the code instead of having to keep some giant file updated.

Reply:
@OzzieOrca Should we make a ticket to investigate this?

Reply:
Ya it's work to do before deploying to production. I haven't captured it in Jira yet.